ucs/adapters/inbound/fastapi_/routes.py

```diff
@@ -0,0 +1,209 @@
+                + " the specified ID because another upload for that file is already"
+                + ' active or has been accepted (its status is not "failed"'
+                + ' "cancelled", or "rejected").',
+            )
+        },
+        status.HTTP_403_FORBIDDEN: {
+            "description": (
+                "The user is not registered as a Data Submitter for the corresponding"
+                + " file."
+            )
+        },
+    },
+)
+def create_upload(upload_creation: UploadCreation):
+    """Initiate a new mutli-part upload for the given file."""
+
+    print(upload_creation)
+    ...
+
+    return ...
+
+
+@router.get(
+    "/uploads/{upload_id}",
+    summary="Get details on a specific upload.",
+    operation_id="getUploadDetails",
+    status_code=status.HTTP_200_OK,
+    response_model=UploadDetails,
+    response_description="Details on a specific upload.",
+    responses={
+        status.HTTP_403_FORBIDDEN: {
+            "description": (
+                "The user is not registered as a Data Submitter for the file"
+                + " corresponding to this multi-part upload."
+            )
+        },
+        status.HTTP_404_NOT_FOUND: {
+            "description": "The multi-part upload with the given ID does not exist."
+        },
+    },
+)
+def get_upload_details(upload_id: str):
+    """
+    Get details on a specific upload.
+    """
+    print(upload_id)
+    ...
+
+
+@router.patch(
+    "/uploads/{upload_id}",
+    summary="Update the status of an existing multi-part upload.",
+    operation_id="updateUploadStatus",
+    status_code=status.HTTP_204_NO_CONTENT,
+    response_description="Multi-part upload successfully updated.",
+    responses={
+        status.HTTP_400_BAD_REQUEST: {
+            "description": (
+                "The provided status value was invalid or the status of this multi-part"
+                + " upload currently or permanently can't be changed."
+            )
+        },
+        status.HTTP_403_FORBIDDEN: {
+            "description": (
+                "The user is not registered as a Data Submitter for the file"
+                + " corresponding to this multi-part upload."
+            )
+        },
+        status.HTTP_404_NOT_FOUND: {
+            "description": "The multi-part upload with the given ID does not exist."
+        },
+    },
+)
+def update_upload_status(upload_id: str, update: UploadUpdate):
+    """
+    Declare a multi-part upload as complete by setting its status to "uploaded".
+    Or cancel a multi-part upload by setting its status to "cancelled".
+    """
+    print(upload_id, update)
+    ...
+
+
+@router.post(
+    "/uploads/{upload_id}/parts/{part_no}/signed_urls",
+    summary="Create new pre-signed URL for a specific part.",
+    operation_id="createPreSignedURL",
+    status_code=status.HTTP_200_OK,
+    response_model=dict,
+    response_description="The newly create pre-signed POST.",
+    responses={
+        status.HTTP_200_OK: {
```

Comment:
I don't quite understand, why we have the 200 Response in Responses here, but not in the other cases

Reply:
Shouldn't we just create a response model for this route asweel?

Reply:
That doesn't really answer my question, why we can't just define a response model. But I just realized, we don't have this defined in the epic docs. We should probably change that there first.

Reply:
Is this basically just "not implemented" for now, in the sense that it would return some kind of dummy object if there was an implementation and needs to be changed later?

Reply:
It is something we forgot in the epic docs. The API call should return a pre-signed upload url for a file part.

I created a PR for this:

https://github.com/ghga-de/epic-docs/pull/28

Reply:
OK, great, I adapted to the proposed solution.